### ISSUE TITLE
KeyError: 'streamingData' When Filtering Audio Streams

### ISSUE THREAD

**Issue by AlienZaki** [JuanBindez/pytubefix#115]
### Describe the Bug
The bug occurs when attempting to filter and retrieve the first audio-only stream from a YouTube video using the `pytubefix` library. Specifically, the error `KeyError: 'streamingData'` is raised, indicating that the key `streamingData` is not found in the `vid_info` dictionary.

### To Reproduce

1. Use the `pytubefix` library to attempt to download audio from a YouTube video.
2. Execute the following code snippet:

   ```python
   audio_stream = yt.streams.filter(only_audio=True).first()
   ```
3. The following error occurs:
   ```
   File "/service/env/lib/python3.8/site-packages/pytubefix/__main__.py", line 502, in streams
       return StreamQuery(self.fmt_streams)
   File "/service/env/lib/python3.8/site-packages/pytubefix/__main__.py", line 226, in fmt_streams
       stream_manifest = extract.apply_descrambler(self.streaming_data)
   File "/service/env/lib/python3.8/site-packages/pytubefix/__main__.py", line 211, in streaming_data
       return self.vid_info['streamingData']
   KeyError: 'streamingData'
   ```

### Expected Behavior

The code is expected to filter the available streams for the YouTube video, retrieve the audio-only streams, and return the first available audio stream without raising an error.

### Desktop

- **OS**: Ubuntu 20.04
- **Python Version**: 3.8
- **Pytubefix Version**:  6.2.2

**Comment by zkreutzjanz** [comment]
Can you try installing pytubefix again (6.3.3)? New release has better error handling.
It will probably print an error message for you to send.

**Comment by AlienZaki** [comment]
I installed pytubefix (6.3.3), I'm getting a different error:
```shell
    |   File "/../service/../transcribe.py", line 31, in download_audio
    |     audio_stream = yt.streams.filter(only_audio=True).first()
    |   File "/../service/env/lib/python3.8/site-packages/pytubefix/__main__.py", line 564, in streams
    |     self.check_availability()
    |   File "/../service/env/lib/python3.8/site-packages/pytubefix/__main__.py", line 324, in check_availability
    |     raise exceptions.VideoPrivate(video_id=self.video_id)
    | pytubefix.exceptions.VideoPrivate: 2Edw66OnUGI is a private video
    +------------------------------------
```
 2Edw66OnUGI is a private video, however it's not!!!

**Comment by JuanBindez** [comment]
I got it normally here, I have no idea what it could be, since it's working here

**Comment by zkreutzjanz** [comment]
I have been able to replicate it. 
On a business account: 
![image](https://github.com/user-attachments/assets/6a695355-e251-4cfd-a236-351efc8755b4)

On a personal account:
![image](https://github.com/user-attachments/assets/db2b0003-50ab-4006-a767-484999f26f91)

**Comment by AlienZaki** [comment]
Yes, I'm getting this:

![Image](https://github.com/user-attachments/assets/0334fef9-8eb4-4d5b-8a82-0e1cbf7d8d90)

It seems I cannot watch any video without login, which is weird.

Also we can see pytubefix raises a wrong error  in this case:
status: LOGIN_REQUIRED
reason: Sign in to confirm you’re not a bot 
it raises `VideoPrivate` which is misleading

**Comment by JuanBindez** [comment]
To better handle these errors we need to generate them and this can sometimes be complicated, so we need the help of everyone who can collaborate by adjusting these errors and making the PR

**Comment by AlienZaki** [comment]
I will try to make my first PR for this misleading error.
Regarding the original error `KeyError: 'streamingData'` it's solved now by upgrading to pytestfix 6.3.3
Thanks.

**Comment by zkreutzjanz** [comment]
> Also we can see pytubefix raises a misleading error in this case:

Agreed

@AlienZaki I am going to go through error handling to clean all this up, I'll make a PR tomorrow or tonight.

See #116 

But right now I am trying to figure out if there is a way around OAuth throttling.
